util/network-devp2p/src/node_table.rs

```diff
@@ -463,6 +487,17 @@ mod tests {
 			node.id);
 	}
 
+	#[test]
+	fn node_parse_fails_for_invalid_urls() {
+		let node = Node::from_str("foo");
+		assert!(node.is_err());
+		assert_matches!(node.unwrap_err().kind(), &ErrorKind::AddressParse);
+
+		let node = Node::from_str("enode://foo@bar");
+		assert!(node.is_err());
+		assert_matches!(node.unwrap_err().kind(), &ErrorKind::AddressParse);
```

Comment:
I believe `assert_matches` is no longer required :)

Reply:
meh, sadly it seems like that's still up in the air: https://github.com/rust-lang-nursery/error-chain/issues/95

assert_matches is used elsewhere, so at least it's not a new-new dependency. :/